room-battle.js

```diff
@@ -318,8 +374,8 @@ class BattleTimer {
 		}
 		let didSomething = false;
 		for (const [slotNum, ticks] of this.turnTicksLeft.entries()) {
-			if (ticks) continue;
-			if (this.settings.timeoutAutoChoose && this.ticksLeft[slotNum] && this.dcTicksLeft[slotNum] === NOT_DISCONNECTED) {
+			if (ticks || (Config.dcTimerBank && this.dcTicksLeft[slotNum])) continue;
```

Comment:
Not sure what your suggesting, this check says:
`If the player has turnTicks left, or the dcTimerBank is enabled AND the player has dcTicks left -> dont disqualify them`
Which is the intended behavior. I'm not sure how not letting turnTicks hit 0 will simplify this.

Reply:
I mean, if the dcTimerBank is enabled AND the player has dcTicks left, `ticks` should always be nonzero. This should be a constraint enforced elsewhere. If that constraint is correctly enforced, this check can just be `ticks`.

Reply:
That is, in fact, how this was previously implemented – the DC timer running out would set `ticks` to `0`, for instance.

Reply:
I scrapped an earlier refactor of this file because I'd been working on it for three days and still wasn't done. I think this would still benefit from a smaller refactor, though.

Reply:
Yeah, I saw how it worked before this refactor. When the dcBank is enabled, a player will use the regular timer when connected, and the DC bank when not. If either runs out, it switches to the other (if there are any ticks left in the other). When both are 0 the player loses.

EX:
player is connected -> runs out of turnTicks -> now using dcTicks -> runs out of dcTicks -> loses
player is connected -> dcs, now using dcTicks -> runs out of dcTicks -> use turnTicks -> those run out -> loses

I'll change the code to not allow turnTicks to hit 0 as long as dcTicks is non-zero (with dcBank enabled). Which should simplify the code as you said.